stmhal/flash.c

```diff
@@ -83,12 +101,27 @@ static const uint32_t flash_info_table[26] = {
     ADDR_FLASH_SECTOR_10, FLASH_SECTOR_10,
     ADDR_FLASH_SECTOR_11, FLASH_SECTOR_11,
     #endif
+	#if defined(FLASH_SECTOR_12)
+	ADDR_FLASH_SECTOR_12, FLASH_SECTOR_12,
+	ADDR_FLASH_SECTOR_13, FLASH_SECTOR_13,
+	ADDR_FLASH_SECTOR_14, FLASH_SECTOR_14,
+	ADDR_FLASH_SECTOR_15, FLASH_SECTOR_15,
+	ADDR_FLASH_SECTOR_16, FLASH_SECTOR_16,
+	ADDR_FLASH_SECTOR_17, FLASH_SECTOR_17,
+	ADDR_FLASH_SECTOR_18, FLASH_SECTOR_18,
+	ADDR_FLASH_SECTOR_19, FLASH_SECTOR_19,
+	ADDR_FLASH_SECTOR_20, FLASH_SECTOR_20,
+	ADDR_FLASH_SECTOR_21, FLASH_SECTOR_21,
+	ADDR_FLASH_SECTOR_22, FLASH_SECTOR_22,
+	ADDR_FLASH_SECTOR_23, FLASH_SECTOR_23,
+	#endif
     ADDR_FLASH_END, 0,
 };
+#define FLASH_INFO_TABLE_SIZE MP_ARRAY_SIZE(flash_info_table)
 
 uint32_t flash_get_sector_info(uint32_t addr, uint32_t *start_addr, uint32_t *size) {
     if (addr >= flash_info_table[0]) {
-        for (int i = 0; i < 24; i += 2) {
+        for (int i = 0; i < FLASH_INFO_TABLE_SIZE-2; i += 2) {
```

Comment:
No need to define FLASH_INFO_TABLE_SIZE, just write MP_ARRAY_SIZE(flash_info_table) directly (also spaces around -).